### ISSUE TITLE
Disparity between console block and eth.syncing

### ISSUE THREAD

**Issue by ADumaine** [ethereumproject/go-ethereum#86]
#### System information

Geth version: 3.0.0 beta
OS & Version: Windows 10
#### Expected behaviour

console should display currentBlock
#### Actual behavior

example from console.  The actual block on etcstats.net at this point was 2,469,954. So there is a discrepancy there too.

```
I1019 22:57:56.729611 core/blockchain.go:973] imported 0 block(s) (0 queued 1 ignored) including 0 txs in 5.0053ms. #2465501 [e3c0d4e2 / e3c0d4e2]
eth.syncing
{
  currentBlock: 2467054,
  highestBlock: 2468887,
  knownStates: 0,
  pulledStates: 0,
  startingBlock: 2467054
}
> I1019 23:05:23.225775 core/blockchain.go:973] imported 1 block(s) (0 queued 0 ignored) including 5 txs in 7m26.4951632s. #2465503 [a3d48836 / a3d48836]
```
#### Background

Was running 1.4.10 and it was periodically falling behind the actual block height.  Updated to 3.0.0 to see if it would help.  Running Win 10 on i7 with 24GB ram.  Ran for 10 hours and it only imported 16 blocks.  The chaindata folder is being populated. geth is using 6.7GB of memory.  Set for 40 peers and 256MB cache. Bandwidth is not an issue (gigabit fiber). The block count is falling further behind.

**Comment by ADumaine** [comment]
So it has been another ten hours and only 46 blocks have been imported.  Now over 5k blocks behind.  I don't understand how this can lag behind while blocks are still being found every 14 seconds and broadcasted to other nodes seemingly no issues according to etcstats.net

```
I1020 08:03:51.980291 core/blockchain.go:973] imported 1 block(s) (0 queued 0 ignored) including 18 txs in 21m13.8008284s. #2465544 [5480f65d / 5480f65d]
I1020 08:03:51.984285 core/blockchain.go:973] imported 0 block(s) (0 queued 1 ignored) including 0 txs in 3.9945ms. #2465545 [0fd9959d / 0fd9959d]
I1020 08:33:01.847591 core/blockchain.go:973] imported 1 block(s) (0 queued 0 ignored) including 13 txs in 29m9.8633062s. #2465546 [18813615 / 18813615]
I1020 08:33:01.851606 core/blockchain.go:973] imported 0 block(s) (0 queued 1 ignored) including 0 txs in 4.0147ms. #2465547 [c174e7e1 / c174e7e1]
I1020 08:33:01.855598 core/blockchain.go:973] imported 0 block(s) (0 queued 1 ignored) including 0 txs in 3.9921ms. #2465548 [cf31dd2b / cf31dd2b]
I1020 08:54:58.150758 core/blockchain.go:973] imported 1 block(s) (0 queued 0 ignored) including 13 txs in 21m56.2951599s. #2465549 [84e83c78 / 84e83c78]
```
Current block in eth.syncing has not changed.
```
> eth.syncing
{
  currentBlock: 2467054,
  highestBlock: 2468887,
  knownStates: 0,
  pulledStates: 0,
  startingBlock: 2467054
}
```
Not sure if this is related but some of the imports say "0 blocks(s)" when it lists a block as being imported.

**Comment by ADumaine** [comment]
Follow up:  gave up on trying to sync with the existing chaindata folder.  Created a new folder, pointed the  --datadir switch to it and used the --fast switch to create new folders and populate.  It looks like it was mostly headers downloaded and it completed in under 2 hours (speed may partly be due to gigabit connection - your download time may be different).  The new folder size is 9.2GB as opposed to the old chaindata folder of 31.3GB.   Geth seems to be keeping in sync and so I should be able to mine again.

It may be possible the original issue was due to either a corrupt database or the fact it was an in-place upgrade from 1.4.10.  
I had tried the upgradedb switch but it tried to upgrade the default folder and would not recognize the --datadir switch.
